Modules/IO/JPEG/src/itkJPEGImageIO.cxx

```diff
@@ -36,37 +42,31 @@ extern "C"
   {
     /* cinfo->err really points to a itk_jpeg_error_mgr struct, so coerce pointer
      */
-    auto * myerr = reinterpret_cast<itk_jpeg_error_mgr *>(cinfo->err);
+    itk_jpeg_error_mgr * myerr = (itk_jpeg_error_mgr *)cinfo->err;
 
     /* Always display the message. */
     /* We could postpone this until after returning, if we chose. */
     (*cinfo->err->output_message)(cinfo);
 
-    jpeg_abort(cinfo); /* clean up libjpeg state */
     /* Return control to the setjmp point */
     longjmp(myerr->setjmp_buffer, 1);
   }
 
-  METHODDEF(void) itk_jpeg_output_message(j_common_ptr) {}
+  METHODDEF(void) itk_jpeg_output_message(j_common_ptr cinfo)
+  {
+#if (defined JPEGIO_JPEG_MESSAGES && JPEGIO_JPEG_MESSAGES == 1)
+    char buffer[JMSG_LENGTH_MAX + 1];
+    (*cinfo->err->format_message)(cinfo, buffer);
+    printf("%s\n", buffer);
```

Comment:
I'd say that printing messages should be done through `itkDebugMacro` macro calls instead of `print`s. But have not checked/tested in which circumstances the message should be printed.

Reply:
IMHO, not, _itkDebugMacro_ is C++ code (`std::ostringstream itkmsg`), i would prefer to avoid C++ code here in `extern "C"`. Also name of function is line number are not informative in particular case.

Reply:
> But have not checked/tested in which circumstances the message should be printed.

The image used in test:
```
Corrupt JPEG data: premature end of data segment
Unsupported marker type 0x02
```
The second from the [post](https://github.com/InsightSoftwareConsortium/ITK/pull/2933#discussion_r767343167)
```
Corrupt JPEG data: 152 extraneous bytes before marker 0xda
Unsupported marker type 0x02
```